qpc/report/merge.py

```diff
@@ -135,6 +138,53 @@ def _merge_json(self):
             print(_(messages.REPORT_JSON_FILES_HELP))
             sys.exit(1)
 
+    def _merge_json_dir(self):
+        """Combine the sources for each json file in a directory.
+
+        :returns Json containing the sources of each file.
+        """
+        report_id = None
+        all_sources = []
+        path = self.args.json_dir[0]
+        if os.path.isdir(path) is not True:
+            print(_(messages.REPORT_JSON_DIR_NOT_FOUND))
+            sys.exit(1)
+        json_files = glob(os.path.join(path, '*.json'))
+        if json_files == []:
+            print(_(messages.REPORT_JSON_DIR_NO_FILES))
+            sys.exit(1)
+        else:
+            print(_(messages.REPORT_JSON_DIR_CHECK_FILES % json_files))
+
+        for file in json_files:
+            with open(file) as lint_f:
+                error = False
+                json_data = json.load(lint_f)
+                try:
+                    # pylint: disable=pointless-statement
+                    json_data['sources']
+                    # pylint: disable=pointless-statement
+                    json_data['sources'][0]['facts']
```

Comment:
You will need to check lengths or you could have an index out of bounds. For example, sources could equal `[]` and index 0 will fail.  I actually would just do this:
```
sources = json_data.get('sources')
if sources:
    facts = sources[0].get('facts)
    if facts:
          facts.get('server_id')
```

This avoids KeyErrors being thrown so you can remove the try/catch.  The checks for sources and facts screen for both None/empty.